doc/doxygenlayout.xml

```diff
@@ -0,0 +1,129 @@
+<doxygenlayout version="1.0">
```

Comment:
I don't think this file is needed when integrating with Sphinx. My recollection is it was to support a native Doxygen website, which we've since retired in libCEED and thus should be safe to remove.